modules/core/src/v2/coins/abstractEthLikeCoin.ts

```diff
@@ -134,15 +151,36 @@ export abstract class AbstractEthLikeCoin extends BaseCoin {
   async signTransaction(
     params: EthSignTransactionOptions,
     callback?: NodeCallback<SignedTransaction>
-  ): Bluebird<SignedTransaction> {
+  ): Bluebird<SignedEthLikeTransaction> {
     const txBuilder = this.getTransactionBuilder();
     txBuilder.from(params.txPrebuild.txHex);
     txBuilder.transfer().key(new Eth.KeyPair({ prv: params.prv }).getKeys().prv!);
     const transaction = await txBuilder.build();
 
+    if (params.isLastSignature) {
+      // In this case when we're doing the second (final) signature
+      return {
+        txHex: transaction.toBroadcastFormat(),
+      };
+    }
+
+    const recipients: Recipient[] = [];
+    let output;
+    for (output of transaction.outputs) {
+      recipients.push({
+        address: output.address,
+        amount: output.value,
+      });
+    }
+
+    const signatures = transaction.signature;
+
     return {
       halfSigned: {
         txHex: transaction.toBroadcastFormat(),
+        signatures: signatures,
```

Comment:
What are these 3 extra fields needed for now that they weren't before?

Reply:
Recipients & Expiration is needed for `explainTransaction`, signatures needed for `signTransaction` (tells us whether isLastSignature should be flagged)